File: global/items/CHRMTimeInZone2.pl
my %whimsy = ();

sub EVENT_ITEM_ENTERZONE {
  $whimsy{$name} = 0; #Initialize player counter
}

sub EVENT_SCALE_CALC {
  if ($whimsy{$name} < 1200) { #120 minutes
    $whimsy{$name}++;
  }
  $questitem->SetScale($whimsy{$name}/1200);
}

File: global/items/CHRMTimeInZone2.lua
-- CHRMTimeInZone2

local time_in_zone = {};

function event_item_enter_zone(e)
	local playerId = e.owner:GetID();
	time_in_zone[playerId] = {1}; -- Start with 1 minutes to avoid divide by 0 (NAN)
end

function event_scale_calc(e)
	local playerId = e.owner:GetID();
	local player_time = 1; -- 1 to prevent NAN

	if time_in_zone[playerId] ~= nil then
			player_time = tonumber(time_in_zone[playerId][1]);

			if player_time == nil then
				-- nil
			elseif player_time< 1200 then -- 2 Hours
					time_in_zone[playerId][1] = player_time + 1;
			end

			e.self:SetScale(tonumber(time_in_zone[playerId][1])/1200);
	else
			time_in_zone[playerId] = {1}; -- Start with 1 minutes to avoid divide by 0 (NAN)
	end
end
